nixos/kubo: make the configuration options idempotent (#199731)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2211.section.md

- PHP 7.4 is no longer supported due to upstream not supporting this
  version for the entire lifecycle of the 22.11 release.

> - The ipfs package and module were renamed to kubo. The kubo module now uses an RFC42-style `settings` option instead of `extraConfig` and the `gatewayAddress`, `apiAddress` and `swarmAddress` options were renamed. Using the old names will print a warning but still work. The `services.kubo.settings` option is now no longer stateful. If you changed any of the options in `services.kubo.settings` (or `services.ipfs.extraConfig`) in the past and then removed them from your NixOS configuration again, those changes are still in your Kubo configuration file but will now be reset to the default. If you're unsure, you may want to make a backup of your configuration file (probably /var/lib/ipfs/config) and compare after the update.

Reviewer: ```suggestion
- The ipfs package and module were renamed to kubo. The kubo module now uses an RFC42-style `settings` option instead of `extraConfig` and the `gatewayAddress`, `apiAddress` and `swarmAddress` options were renamed. Using the old names will print a warning but still work. 

- The `services.kubo.settings` option is now no longer stateful. If you changed any of the options in `services.kubo.settings`/`services.ipfs.extraConfig` in the past and then removed them from your NixOS configuration again, those changes are still in your Kubo configuration file but will now be reset to the default. If you're unsure, you may want to make a backup of your configuration file (most likely /var/lib/ipfs/config) and compare after the update.
```

Having this in one line is way to long for my reading liking.

Author: I had to move this into the the 23.05 release notes anyways.
